pandas/_typing.py

```diff
@@ -447,7 +447,7 @@ def closed(self) -> bool:
 AnyAll = Literal["any", "all"]
 
 # merge
-MergeHow = Literal["left", "right", "inner", "outer", "cross"]
+MergeHow = Literal["left", "right", "inner", "outer", "cross", "leftsemi"]
```

Comment:
`MergeHow` is also used as the type for `how` in `DataFrame.join()`.  

So if `DataFrame.join()` will not support the `"leftsemi"` operation, then you'll need to split this type into a `JoinHow` and `MergeHow`, with the latter dependent on the former.

If `DataFrame.join()` will support that operation, then you'll have to update the docs for `join` and add a test for it.